### PR TITLE
STORM-2416: Improve Release Packaging to Reduce File Size

### PR DESCRIPTION
**Pull request by ptgoetz**
This brings the binary distribution file size down from > 200MB to ~79MB.

### CODE REVIEW

**Comment by vesense** [comment]
+1

File: storm-dist/binary/src/main/assembly/binary.xml [hunk 1]
```diff
@@ -77,11 +77,14 @@
                 <exclude>js/script.js</exclude>
             </excludes>
         </fileSet>
+        <!-- EXAMPLES -->
         <fileSet>
             <directory>${project.basedir}/../../examples</directory>
             <outputDirectory>examples</outputDirectory>
             <excludes>
                 <exclude>**/target/**</exclude>
+                <exclude>**/dependency-reduced-pom.xml</exclude>
```
Comment (HeartSaVioR) [inline storm-dist/binary/src/main/assembly/binary.xml]: Nice catch for two lines! :)

File: examples/storm-perf/pom.xml [hunk 2]
```diff
@@ -22,8 +22,8 @@
     <parent>
         <artifactId>storm</artifactId>
         <groupId>org.apache.storm</groupId>
-        <version>1.1.1-SNAPSHOT</version>
-        <relativePath>..</relativePath>
+        <version>1.1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (HeartSaVioR) [inline examples/storm-perf/pom.xml]: Maybe `../../pom.xml`?
(given that storm-perf compile is failing in Travis build... https://travis-ci.org/apache/storm/jobs/212268892)

Comment (HeartSaVioR) [inline examples/storm-perf/pom.xml]: @ptgoetz 
Fixing this to `../../pom.xml` resolved the error on `mvn clean install -DskipTests`. Please address this.

**Comment by HeartSaVioR** [comment]
Looks great otherwise. I'd be +1 once build issue is fixed.

**Comment by HeartSaVioR** [comment]
@ptgoetz 
Could you make the change for master branch? Maybe 1.0.x branch too.

File: examples/storm-perf/pom.xml (hunk 2 above)
Comment (ptgoetz) [inline examples/storm-perf/pom.xml]: @HeartSaVioR Nice catch, fixed.

**Comment by ptgoetz** [comment]
@HeartSaVioR Yes, that's my plan. I prioritized the 1.1.x branch since it's needed for the 1.1.0 release.

I'm not sure we want to apply it to the 1.0.x branch since we have a number of releases from that branch and this change would change the structure of the distribution. I could go either way, though, and am open to others' opinions.

**Comment by HeartSaVioR** [comment]
@ptgoetz 
OK either applying or not applying this to 1.0.x is fine for me. 
Given that you're planning to apply this to master branch as well, I think this is mergeable now.
+1

**Pull request merged**